pyccel/ast/core.py

```diff
@@ -2930,29 +2978,39 @@ class Interface(PyccelAstNode):
     is_argument : bool
         True if the interface is used for a function argument.
 
+    is_imported : bool
+        True if the interface is imported from another file.
+
+    syntactic_node : FunctionDef, default: None
+        The syntactic node that is not annotated.
+
     Examples
     --------
     >>> from pyccel.ast.core import Interface, FunctionDef
     >>> f = FunctionDef('F', [], [], [])
     >>> Interface('I', [f])
     """
-    __slots__ = ('_name','_functions','_is_argument')
+    __slots__ = ('_name','_functions','_is_argument', '_is_imported', '_syntactic_node')
     _attribute_nodes = ('_functions',)
 
     def __init__(
         self,
         name,
         functions,
         is_argument = False,
+        is_imported=False,
+        syntactic_node=None,
         ):
 
         if not isinstance(name, str):
             raise TypeError('Expecting an str')
-        if not isinstance(functions, list):
+        if not isinstance(functions, (list, tuple)):
             raise TypeError('Expecting a list')
         self._name = name
         self._functions = functions
```

Comment:
If `functions` is a generic iterable and we want a tuple (for example), than it is sufficient to write
```suggestion
        self._functions = tuple(functions)
```

Reply:
[681b62d](https://github.com/pyccel/pyccel/pull/1728/commits/681b62dbc14f0f67b29d47d90e8869e1d9ad03f5)